openshift-kube-apiserver/admission/customresourcevalidation/infrastructure/validate_infrastructure.go

```diff
@@ -0,0 +1,133 @@
+package infrastructure
+
+import (
+	"fmt"
+	"io"
+	"regexp"
+
+	apiv1 "github.com/openshift/api/config/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/kubernetes/openshift-kube-apiserver/admission/customresourcevalidation"
+	"k8s.io/kubernetes/pkg/apis/core/validation"
+)
+
+const (
+	PluginName = "config.openshift.io/ValidateInfrastructure"
+)
+
+type infrastructureV1 struct{}
+
+// kubernetesNamespaceRegex is used to check that a tag key is not in the kubernetes.io namespace.
+var kubernetesNamespaceRegex = regexp.MustCompile(`^([^/]*\.)?kubernetes.io/`)
```

Comment:
does this also stop "kubernetes.io", "kubernetes.io/"?  Can I get a unit test?

Reply:
Hi @deads2k 
This just stops kubernetes.io/, from they way I understood our needs.
I can for sure update it to catch kubernetes.io and openshift.io as well

Reply:
It is unclear to me what the validation is supposed to be: 
* The CRD and the enhancement only state that the tag name cannot contain a `kubernetes.io` or `openshift.io` **prefix**.
* The commit text states that the tag name cannot contain a `kubernetes.io` or `openshift.io` **suffix**.
* The PR description state that the tag name cannot contain a `kubernetes.io` or `openshift.io` **at all**.

Reply:
Hi @sanchezl 
Thanks for reviewing!
You are right about this, it reflects the discussions we've had with the api team the last few days.
The final decision was to forbid tag names containing  `kubernetes.io` or `openshift.io`.
I can update the commit text when squashing commits
@TrilokGeer do we need to update the enhancement?

Reply:
Git commit message updated